core/commands/cherry_pick.py

```diff
@@ -1,13 +1,40 @@
 import sublime
+from sublime_plugin import TextCommand
 
 from .log import GsLogCommand
 from ..git_mixins.branches import BranchesMixin
+from ...common import util
+
+
+class GsCherryPickHashCommand(TextCommand, BranchesMixin):
+
+    """
+    Cherry-pick a specific hash.
+    """
+
+    def run(self, edit):
+        lines = util.view.get_lines_from_regions(self.view, self.view.sel())
+
+        hashes = list(
+            line for line in list(
+                line.strip(" /_\|*").split(' ')[0] for line in lines
+            ) if line)
+
+        if not len(hashes) == 1:
+            sublime.status_message("You can only cherry-pick one commit at a time.")
+            return
+
+        self.view.window().run_command("gs_cherry_pick", {"target_hash": hashes[0]})
+        # This line doest always works correct. We want to refresh the
+        # view after cherry-picking it, but the cherry-pick command is
+        # async so this method get called before the cherry-pick is done
+        self.view.run_command("gs_branches_diff_branch_history_refrech")
```

Comment:
Here's what we should do:

- This line should be removed.
- Then, [at the end of `GsCherryPickCommand`](https://github.com/divmain/GitSavvy/blob/be3f6afc3d8955567b3842787fe769533cb43ad8/core/commands/cherry_pick.py#L39-L39), we should add [`util.view.refresh_gitsavvy(self.window.active_view())`](https://github.com/divmain/GitSavvy/blob/be3f6afc3d8955567b3842787fe769533cb43ad8/core/commands/pull.py#L92-L92).
- Finally, modify [`refresh_gitsavvy`](https://github.com/divmain/GitSavvy/blob/be3f6afc3d8955567b3842787fe769533cb43ad8/common/util/view.py#L61-L68) so that it checks whether the active view is a branch history view and, if so, it should run `gs_branches_diff_branch_history_refresh`.